extension/lib/login.js

```diff
@@ -0,0 +1,101 @@
+const BigCommerce = require('node-bigcommerce')
+const request = require('request-promise-native')
+const get = require('lodash.get')
+const InvalidCredentialsError = require('./errors/InvalidCredentialsError')
+const BigCommerceCustomerRepository = require('./bigcommerce/CustomerRepository')
+
+let customerRepo
+
+/**
+ * @param {PipelineContext} context
+ * @param {LoginInput} input
+ * @returns {Promise<LoginResponse>}
+ */
+module.exports = async (context, input) => {
+  if (!customerRepo) {
+    customerRepo = BigCommerceCustomerRepository.create(new BigCommerce({
+      logLevel: 'info',
+      clientId: context.config.clientId,
+      accessToken: context.config.accessToken,
+      storeHash: context.config.storeHash,
+      responseType: 'json',
+      apiVersion: 'v2'
+    }))
+  }
+
+  try {
+    const success = await submitLoginForm(
+      `${input.shopUrl}/login.php?action=check_login`,
+      input.parameters.login,
+      input.parameters.password
+    )
+    if (!success) {
```

Comment:
There is a flow issue indication on this point. It practically means this error should not be thrown here but rather inside of the function submitLoginForm.

Reply:
I've reformatted the function so that it rather throws the error then it returns a value. I think it is better encapsulated now.